pallets/runtime/tests/src/asset_test.rs

```diff
@@ -2498,20 +2500,23 @@ fn issuers_can_redeem_tokens_from_portfolio_with_custodian() {
                 token.total_supply
             );
 
+            // Add auth for custody to be moved to bob
             let auth_id = Identity::add_auth(
                 owner.did,
                 Signatory::from(bob.did),
                 AuthorizationData::PortfolioCustody(user_portfolio),
                 None,
             );
 
+            // Check that bob accepts auth
             assert_ok!(Portfolio::accept_portfolio_custody(bob.origin(), auth_id));
 
             assert_eq!(
                 Portfolio::portfolio_custodian(user_portfolio),
                 Some(bob.did)
             );
 
+            // Check error is given when previous custodian tries to redeem form portfolio
             assert_noop!(
                 Asset::redeem_from_portfolio(
                     owner.origin(),
```

Comment:
Dave (`owner`) isn't the "previous custodian".  Dave is still the owner of the portfolio.

Portfolio custodians can only manage the assets in that portfolio for the owner.  The custodian doesn't own those assets.